### DISCUSSION TITLE
Need to manage the state between two different root-config hosts. (9000 and 9090)

### DISCUSSION THREAD

**Discussion by Anand-philips** [community/community#45121] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I was using the one launchpad root config host 9000 where I have one onclick event which will route to another root config host 9090. I need to manage the state between those two root config app. Both has one app has common (nav Bar App). please refer attachment for more details. please check me with your suggestion. Many Thanks in advance. 
<img width="952" alt="Capture2" src="https://user-images.githubusercontent.com/122438240/213984630-8aab0533-5478-42c9-bea7-d7abcffa89c1.PNG">
![InkedCapture1](https://user-images.githubusercontent.com/122438240/213984636-c1e63015-188b-40af-bec8-d867847cfd07.jpg)

**Comment by jge162** [comment]
@Anand-philips a option you can use is to use a state management library like Redux or MobX to manage the state of your application. This will allow you to share the state between the two applications and ensure that the navigation bar component is updated with the correct state when routing between the two applications.

**Reply by Anand-philips** [reply]
Yes, Currently I'm using the redux micro frontend (https://github.com/microsoft/redux-micro-frontend) method with Single Spa. I can manage the global state within the same container that is within the same domain like localhost:9000 but when I'm moving from localhost:9000 to localhost 9090 another root config app thru on click event.     The global store in 9000 is destroyed and I can't find any data from the 9000 config app.
For your reference, the navBar app is the only one common for both root config Apps. I'm trying to change the header state on the NavBar app in 9000 hosts and I want to get the changed state in another root config app 9090.

Please help me how I can get the state management between two (9000 and 9090) root config App. Thanks for your Time and Support.

**Reply by jge162** [reply]
@Anand-philips A possible solution to this problem would be to use a centralized state management like the one your using now Redux, Redux allows you to store and manage your application's state in a single location. Instead of each root config app having its own separate store, you can use a single store shared by all the apps. This way, when you update the state in one app, it will be reflected in all other apps that are connected to the same store. 

You can use middleware like 
redux-thunk https://github.com/reduxjs/redux-thunk

or 

redux-saga https://github.com/redux-saga/redux-saga

to handle the async actions that dispatches to update the state from one root config app to another.

**Comment by yasraubi** [comment]
se
